### PR TITLE
First pass at license enforcement update

### PR DESCRIPTION
**Pull request by hardillb**
part of FlowFuse/flowfuse#4074

## Description

Describes the changes to what happens with expired licenses from v2.8.0 onward

## Related Issue(s)

FlowFuse/flowfuse#4074

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] I have considered the performance impact of these changes
 - [ ] Suitable unit/system level tests have been added and they pass
 - [ ] Documentation has been updated

### CODE REVIEW

**Comment by hardillb** [comment]
Do not merge until 2.8.0 release

File: src/changelog/2024/08/enterprise-license-update.md
Comment (hardillb) [inline src/changelog/2024/08/enterprise-license-update.md:18]:
```suggestion
If you are currently running with older version with  an expired license
```

File: src/changelog/2024/08/enterprise-license-update.md
Comment (hardillb) [inline src/changelog/2024/08/enterprise-license-update.md:18]:
```suggestion
If you are currently running with older version with an expired license
```

File: src/changelog/2024/08/enterprise-license-update.md
Comment (hardillb) [inline src/changelog/2024/08/enterprise-license-update.md:42]:
```suggestion
- No new Instances or Devices can be created.
- A email will be sent to all Admin users saying the license has expired.
- A banner showing the license as expired will be shown to all users.

   ![expired banner](images/expired-banner.png)
```

File: src/changelog/2024/08/enterprise-license-update.md
Comment (ZJvandeWeg) [inline src/changelog/2024/08/enterprise-license-update.md:22]: Do changelogs allow for a "fold"? Do we need a `<!--more-->` here?

File: src/changelog/2024/08/enterprise-license-update.md
Comment (ZJvandeWeg) [inline src/changelog/2024/08/enterprise-license-update.md:21]: I'm not a fan of the royal we: (hope I kept the 80 char limit)

```suggestion
Starting with the v2.8.0 release FlowFuse will be taking a slightly stricter 
approach to License enforcement outside of FlowFuse Cloud. Previously
when a license expired the platform would continue to run with the same
capabilities as when the license was valid.

If you are currently running with older version with an expired license
please [contact the Sales team](/contact-us/) to discuss renewing your
license before upgrading. You can check your current license on the 
Admin Settings -> Overview page.
```

File: src/changelog/2024/08/enterprise-license-update.md
Comment (ZJvandeWeg) [inline src/changelog/2024/08/enterprise-license-update.md:37]:
```suggestion
- Instances will not be started.
```

File: src/changelog/2024/08/enterprise-license-update.md
Comment (ZJvandeWeg) [inline src/changelog/2024/08/enterprise-license-update.md:50]: When applying a license, should the application be restarted? https://flowfuse.com/docs/upgrade/open-source-to-premium/ suggests it wouldn't

**Review by ZJvandeWeg — CHANGES_REQUESTED** [review]

File: src/changelog/2024/08/enterprise-license-update.md
Comment (hardillb) [reply src/changelog/2024/08/enterprise-license-update.md:50]: We don't have automatic restart this time round (we also don't have a flag for it in the db). So users will have to manually restart instances.

File: src/changelog/2024/08/enterprise-license-update.md
Comment (hardillb) [reply src/changelog/2024/08/enterprise-license-update.md:22]: Changelogs don't normally have a fold from the previous examples

File: src/changelog/2024/08/enterprise-license-update.md
Comment (ZJvandeWeg) [reply src/changelog/2024/08/enterprise-license-update.md:50]: I wasn't clear, I'm sorry. The app server, fuse app, doesn't need to be restarted?

**Comment by ZJvandeWeg** [comment]
@joepavitt can you merge when this is ready?

File: src/changelog/2024/08/enterprise-license-update.md
Comment (hardillb) [reply src/changelog/2024/08/enterprise-license-update.md:50]: A restart is only needed for the first time a license is applied, if the license has just expired updating it via the web interface is sufficient as all the feature flags are already loaded.

**Comment by hardillb** [comment]
@ZJvandeWeg I think you will need to re-review to remove the changes requested flag before we can merge even if Joe approves

File: src/changelog/2024/08/enterprise-license-update.md
Comment (joepavitt) [inline src/changelog/2024/08/enterprise-license-update.md:18]:
```suggestion
If you are currently running with an older version with an expired license
```

**Review by joepavitt — APPROVED** [review]

**Pull request merged**
